`plat/xen`: Build fails on ARM64 with networking enabled: `‘GRANT_INVALID_REF’ undeclared`

### Describe the bug

Building applications using networking for Xen ARM64 results in a build error.

### Steps to reproduce

Build C HTTP for Xen/ARM64 in the [`catalog-core` repository](https://github.com/unikraft/catalog-core). Follow the steps:

1. Clone the repository:

   ```console
   git clone https://github.com/unikraft/catalog-core
   ```

1. Enter the `c-http/` directory:

   ```console
   cd catalog-core/c-http/
   ```
   
1. Switch to the `scripts` branch:

   ```console
   git checkout scripts
   ```

1. Build C HTTP for Xen/ARM64:

   ```console
   ./build.xen.arm64
   ```

   The build error occurs.

### Expected behavior

Build is successful, without errors.

### Which architectures were you using or does this bug affect?

x86_64, arm64

### Which platforms were you using or does this bug affect?

xen

### Relevant log output

```shell
AS      libxenconsole: libinfo.libuklibid.o
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c: In function ‘netfront_rxq_enqueue’:
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:241:39: error: ‘GRANT_INVALID_REF’ undeclared (first use in this function)
  241 |         if (unlikely(rxq->gref[id] == GRANT_INVALID_REF)) {
      |                                       ^~~~~~~~~~~~~~~~~
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/include/uk/arch/lcpu.h:53:43: note: in definition of macro ‘unlikely’
   53 | #define unlikely(x) (__builtin_expect((!!(x)), 0))
      |                                           ^
  LD      libxenplat.ld.o
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c: In function ‘netfront_txq_setup’:
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:468:36: error: ‘GRANT_INVALID_REF’ undeclared (first use in this function)
  468 |         UK_ASSERT(txq->ring_ref != GRANT_INVALID_REF);
      |                                    ^~~~~~~~~~~~~~~~~
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/include/uk/arch/lcpu.h:53:43: note: in definition of macro ‘unlikely’
   53 | #define unlikely(x) (__builtin_expect((!!(x)), 0))
      |                                           ^
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:468:9: note: in expansion of macro ‘UK_ASSERT’
  468 |         UK_ASSERT(txq->ring_ref != GRANT_INVALID_REF);
      |         ^~~~~~~~~
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:477:25: warning: implicit declaration of function ‘gnttab_end_access’ [-Wimplicit-function-declaration]
  477 |                         gnttab_end_access(txq->ring_ref);
      |                         ^~~~~~~~~~~~~~~~~
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c: In function ‘netfront_rxq_setup’:
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:547:36: error: ‘GRANT_INVALID_REF’ undeclared (first use in this function)
  547 |         UK_ASSERT(rxq->ring_ref != GRANT_INVALID_REF);
      |                                    ^~~~~~~~~~~~~~~~~
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/include/uk/arch/lcpu.h:53:43: note: in definition of macro ‘unlikely’
   53 | #define unlikely(x) (__builtin_expect((!!(x)), 0))
      |                                           ^
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/drivers/xen/net/netfront.c:547:9: note: in expansion of macro ‘UK_ASSERT’
  547 |         UK_ASSERT(rxq->ring_ref != GRANT_INVALID_REF);
```

--- a/drivers/xen/net/Config.uk
+++ b/drivers/xen/net/Config.uk
@@ -1,6 +1,7 @@
 config LIBXEN_NETFRONT
 	bool "Xenbus Netfront Driver"
 	select LIBXEN_XENBUS
+	depends on XEN_GNTTAB
 	depends on PLAT_XEN
 	depends on LIBUKNETDEV
 	help

--- a/plat/xen/Config.uk
+++ b/plat/xen/Config.uk
@@ -44,7 +44,7 @@ config XEN_PV_BUILD_P2M
 
 config XEN_GNTTAB
 	bool "Grant table support"
-	default y if XEN_PV
+	default y if (XEN_PV || ARCH_ARM_64)
 	depends on (ARCH_X86_64 || ARCH_ARM_64)
 	select LIBXEN_XENBUS
 	select LIBUKALLOC

